debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/RowDeserializers.java

```diff
@@ -322,19 +322,40 @@ protected static Serializable deserializeTimeV2(int meta, ByteArrayInputStream i
          * + fractional-seconds storage (size depends on meta)
          */
         long time = bigEndianLong(inputStream.read(3), 0, 3);
-        int hour = bitSlice(time, 2, 10, 24);
-        int minute = bitSlice(time, 12, 6, 24);
-        int second = bitSlice(time, 18, 6, 24);
-        int nanoSeconds = deserializeFractionalSecondsInNanos(meta, inputStream);
-        return LocalTime.of(hour, minute, second, nanoSeconds);
+        boolean is_negative = bitSlice(time, 0, 1, 24) == 0;
+        int hours = bitSlice(time, 2, 10, 24);
+        int minutes = bitSlice(time, 12, 6, 24);
+        int seconds = bitSlice(time, 18, 6, 24);
+        int nanoSeconds;
+        if (is_negative) { // mysql binary arithmetic for negative encoded values
+            int mask10bits = (1 << 10) - 1;
```

Comment:
Seems we could extract those two as constants?

Reply:
Will do

Reply:
I'll leave it to you. Having some unit tests would have been nice, but as you say we already got it covered in the ITs, so it seems not strictly needed.